src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ParameterAssignmentCheck.java

```diff
@@ -178,7 +189,8 @@ public void leaveToken(DetailAST ast) {
         switch (ast.getType()) {
             case TokenTypes.CTOR_DEF:
             case TokenTypes.METHOD_DEF:
-                leaveMethodDef();
+            case TokenTypes.LAMBDA:
+                leaveMethodDefOrLambda();
```

Comment:
please rename

Reply:
renamed to `popParameterNamesStack`, can rename to `popParameterNamesDeque` but since it is a pop operation I think stack will be better, we use Deque just 'cause it's unsynchronized so stack shall be fine but  I'm open to changing it.